engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -519,6 +520,52 @@ public static FileSystem newFileSystemInstance(Configuration conf) throws IOExce
     }
   }
 
+  /**
+   * This gets the DFS Root dir to be used by Apex applications as per the following logic:
+   *    - if Apex-root-dir is null then just call getDTDFSRootDir to get the old behavior
+   *    - if apex.dfsRootDirectory defined use it, otherwise fall back to using dt.dfsRootDirectory by calling getDTDFSRootDir.
+   *    - Value of apex.dfsRootDirectory is referred to as Apex-root-dir below.
+   *    - if Apex-root-dir value is an absolute path then scan for %USER_NAME% in the string and replace it with the user's name, else use the absolute path as is.
+   *    - if Apex-root-dir value is a relative path then append it to the user's home directory.
+   *
+   * In the above steps a "user" refers to either impersonating or impersonated user.
+   * If the value of apex.application.path.impersonated is true we will use the impersonated user else impersonating user.
+   *
+   * @param fs
+   * @param conf
+   * @param impersonated user
```

Comment:
Can you check javadoc formatting and any missing information like argument description.